### PR TITLE
feat: add hcaptcha to register and forgot password

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/6086

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: config/environment.js [hunk 1]
```diff
@@ -79,6 +79,12 @@ module.exports = function(environment) {
     authorizer: 'authorizer:jwt'
   };
 
+  ENV['ember-h-captcha'] = {
+    jsUrl   : 'https://hcaptcha.com/1/api.js', // default
+    sitekey : '4d94db5b-d3fd-407b-a574-e6ba0af9b73c',
```
Comment (iamareebjamal) [inline config/environment.js:84]: Hard-coded key

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by iamareebjamal** [comment]
If key is not present, captcha shouldn't be enabled.

**Comment by maze-runnar** [comment]
so key would be added by .env or from admin section?

**Comment by iamareebjamal** [comment]
.env

**Comment by mariobehling** [comment]
We have an area for a Google captcha in the admin section. Ultimately it would be good to have it here.

Screenshot from 2020-12-19 20-43-54

**Comment by iamareebjamal** [comment]
@mariobehling GSoC students just added it here for show without implementing it. We need the key at build time for the captcha service, just like the Google Maps API key, thus it is not possible to fetch it at runtime

File: .env.example [hunk 2]
```diff
@@ -1,4 +1,5 @@
 GOOGLE_API_KEY="Sample Key"
+HCAPTCHA_SITE_KEY=""
```
Comment (iamareebjamal) [inline .env.example:2]: Move it to last and comment out

**Comment by iamareebjamal** [comment]
Captcha shouldn't be enabled if key isn't present, as reviewed previously

File: app/templates/components/forms/register-form.hbs [hunk 3]
```diff
@@ -37,8 +37,9 @@
           </div>
         </div>
       </div>
+      <HCaptcha @onSuccess={{action 'captchaSuccess'}} @onExpired={{action 'onCaptchaExpired'}}/>
```
Comment (iamareebjamal) [inline app/templates/components/forms/register-form.hbs:40]: Simply use mut action instead of defining 2 actions in js

**Comment by maze-runnar** [comment]
> Captcha shouldn't be enabled if key isn't present, as reviewed previously

ok

File: app/components/forms/register-form.js [hunk 4]
```diff
@@ -53,6 +56,10 @@ export default Component.extend(FormMixin, {
     }
   },
 
+  showHcaptcha: computed(function() {
+    return ENV.hcaptchaKey === null || ENV.hcaptchaKey === undefined ? false : true;
```
Comment (iamareebjamal) [inline app/components/forms/register-form.js:60]:
```suggestion
    return !!ENV.hcaptchaKey;
```

Comment (iamareebjamal) [reply app/components/forms/register-form.js:60]: This doesn't change so no need of computed

File: app/components/forms/register-form.js [hunk 5]
```diff
@@ -6,6 +6,7 @@ export default Component.extend(FormMixin, {
   email     : '',
   password  : '',
   isLoading : false,
+  captcha   : false,
```
Comment (Jyotsna1006) [inline app/components/forms/register-form.js]: can i handle the issue?

Comment (maze-runnar) [reply app/components/forms/register-form.js]: what...?

**Pull request merged**